sop-core/sop-core.cabal

```diff
@@ -77,3 +77,11 @@ library
                        EmptyCase
                        UndecidableSuperClasses
                        BangPatterns
+
+test-suite doctests
```

Comment:
Note: this works only for `sop-core` which doesn't have extra dependencies. In other words this approach (without `build-type: Custom` and `cabal-docspec` like thing) won't work for `generics-sop`.